robot/basestation/static/js/elements/navigation/navigation.js

```diff
@@ -1,191 +1,661 @@
 $(document).ready(() => {
+  goalList = []
+  goalCount = 0
+
+  let minuteTodecmial = 60
+  let secondTodecmial = 3600
+
   // setup a subscriber for the rover_position topic
   let rover_position_listener = new ROSLIB.Topic({
-    ros: ros,
-    name: 'rover_position',
-    messageType: 'geometry_msgs/Point'
+      ros: ros,
+      name: 'rover_position',
+      messageType: 'geometry_msgs/Point'
   })
-  rover_position_listener.subscribe(function (message) {
-    $('#rover-latitude').text(message.x)
-    $('#rover-longitude').text(message.y)
-    $('#rover-heading').text(message.z)
+  rover_position_listener.subscribe(function(message) {
+      $('#rover-latitude').text(message.x)
+      $('#rover-longitude').text(message.y)
+      $('#rover-heading').text(message.z)
   })
   // setup a subscriber for the antenna_goal topic
   let antenna_goal_listener = new ROSLIB.Topic({
-    name: 'antenna_goal',
-    messageType: 'geometry_msgs/Point',
-    ros: ros
+      name: 'antenna_goal',
+      messageType: 'geometry_msgs/Point',
+      ros: ros
+  })
+  antenna_goal_listener.subscribe(function(message) {
+      $('#antenna-stats-rec-angle').text(parseFloat(message.x).toFixed(3))
+      $('#antenna-stats-dist-to-rover').text(parseFloat(message.y).toFixed(2))
   })
-  antenna_goal_listener.subscribe(function (message) {
-    $('#recommended-antenna-angle').text(parseFloat(message.x).toFixed(3))
-    $('#distance-to-rover').text(parseFloat(message.y).toFixed(2))
+  // setup topics to communicate with GoalsNode
+  let create_goal_publisher = new ROSLIB.Topic({
+      ros: ros,
+      name: 'create_goal',
+      messageType: 'mcu_control/RoverGoal'
+  })
+  let delete_goal_publisher = new ROSLIB.Topic({
+      ros: ros,
+      name: 'delete_goal',
+      messageType: 'std_msgs/String'
+  })
+  
+  goal_list_subscriber.subscribe(function(message) {
+    if (message.goal_list.length > 0) {
+        goal_latitude.set(message.goal_list[0].latitude)
+        goal_longitude.set(message.goal_list[0].longitude)
+        $('#goal-stats-latitude').text(message.goal_list[0].latitude.toFixed(6))
+        $('#goal-stats-longitude').text(message.goal_list[0].longitude.toFixed(6))
+    } else {
+        goal_latitude.set(false)
+        goal_longitude.set(false)
+        $('#goal-stats-latitude').text('----')
+        $('#goal-stats-longitude').text('----')
+    }
+
+    if (goalList.length != message.goal_list.length) {
+      if (goalList.length < message.goal_list.length) {
+        goalCount++
+      }
+
+      goalList = message.goal_list
+
+      untoggleGoals()
+      toggleGoals()
```

Comment:
what's this for?

Reply:
This was just a small brain solution to an issue that was meant to be temporary, oops. The untoggleGoals function was to delete all of the goals in the "add goals" window. I removed that function and just kept the toggleGoals function in order to bring changes to the gui for when the goal list is updated.